### ISSUE TITLE
Alternatives to Gurobi

### ISSUE THREAD

**Issue by shubhamwagh** [art-programmer/FloorNet#18]
Hello!
It seems that Gurobi is being used which is not completely free. Any alternatives to evaluate the model without the use of Gurobi? It would have been better if this dependency is mentioned on the ReadMe file as well.

**Comment by art-programmer** [comment]
@shubhamwagh I just replicate the IP optimization using a free solver. The interface is same as before. Please look at IP.py for more details.

**Comment by shubhamwagh** [comment]
That's really good. I will definitely take a look at it after holidays.

**Comment by shubhamwagh** [comment]
I was evaluating the performance of the trained model. I get an error 
`NUM_ICONS is not defined in IP.py`
Is that the total number of icons in the icon folder?

**Comment by shubhamwagh** [comment]
Two things not included:
1) from floorplan_utils.py import *  -> is not included in IP.py script
2) findMatches function is not included -> which I coped it from QP.py file

Hope this helps some one who will use it.

**Comment by shubhamwagh** [comment]
@art-programmer  there are lot of bugs in IP.py like NUM_ROOMS has been used instead of NUM_FINAL_ROOMS and similarly for icons etc. This creates index pout of bounds errors. It will be better if you can rectify it.

**Comment by shubhamwagh** [comment]
Fixed the bugs, atleast I am able to evaluate the trained model with the provided dataset.

**Comment by Muhammad-Ahmad-Waseem** [comment]
> Fixed the bugs, atleast I am able to evaluate the trained model with the provided dataset.

Can you kindly share the bug-free IP.py file or guide me on how to solve the index pout of bounds errors?

**Comment by AurelaShehu** [comment]
This worked for me:

[IP.zip](https://github.com/art-programmer/FloorNet/files/5532071/IP.zip)

and in evaluate.py I have:

import numpy as np
import cv2
#default
#from QP import reconstructFloorplan, findMatches
from QP import findMatches
from IP import reconstructFloorplan
from RecordReader import *
from train import *

**Comment by KirillHiddleston** [comment]
> This worked for me:
> 
> [IP.zip](https://github.com/art-programmer/FloorNet/files/5532071/IP.zip)
> 
> and in evaluate.py I have:
> 
> import numpy as np
> import cv2
> #default
> #from QP import reconstructFloorplan, findMatches
> from QP import findMatches
> from IP import reconstructFloorplan
> from RecordReader import *
> from train import *

Yes its working but u need install pulp 2.0 pip install pulp==2.0
